Source/bmalloc/libpas/src/libpas/pas_local_allocator.c

```diff
@@ -242,6 +242,40 @@ bool pas_local_allocator_stop(
     allocator->scavenger_data.is_in_use = true;
     pas_compiler_fence();
 
+    /* The scavenger thread may race with the client thread on calling pas_local_allocator_stop.
+       The scavenger will first suspend the client thread before checking the is_in_use flag.
+       If is_in_use is set, the scavenger will not call pas_local_view_cache_stop, and move on.
+       If is_in_use is not set, the scavenger will call pas_local_allocator_scavenger_data_stop,
+       which in turn calls pas_local_allocator_stop to stop the local_allocator.
+
+       Normally, if the scavenger has already completed its call to pas_local_allocator_stop
+       before the client calls it, pas_local_allocator_stop will just return early for the client.
+       This is because pas_local_allocator_stop first check pas_local_allocator_scavenger_data_is_stopped
+       before doing the work to stop the local_allocator.
+
+       However, if the client thread is already in the process of executing pas_local_allocator_stop,
+       gets passed the pas_local_allocator_scavenger_data_is_stopped check, and then, gets suspended by
```

Comment:
nit: past

Reply:
I think "past" is correct.  I initially wrote it as "pass".  Saam corrected it to "passed" because I wrote "pass".  But what I actually meant is "past" as in the pas_local_allocator_scavenger_data_is_stopped check already occurred.